test/capi/capi_sframe.cxx

```diff
@@ -1503,95 +1501,151 @@ void test_sframe_read_json() {
+  TS_ASSERT(error == NULL);
+
+  tc_flex_list_add_element(column_list, user_id_ft, &error);
+  TS_ASSERT(error == NULL);
+
+  tc_groupby_aggregator_add_count(gb_manual, "count", &error);
+  TS_ASSERT(error == NULL);
+
+  tc_sframe* sampled_frame = tc_sframe_group_by(sf, column_list, gb_manual, &error);
+  TS_ASSERT(error == NULL);
+
+  // make groupkeys and operators, call C++ groupby
+  std::vector<std::string> group_keys;
+  group_keys.push_back("user_id");
+  std::map<std::string, turi::aggregate::groupby_descriptor_type> operators;
+  operators.insert({"count", turi::aggregate::COUNT()});
+
+  turi::gl_sframe sampled_gl_sframe = sf_gl.groupby(group_keys, operators);
+
+  TS_ASSERT(check_equality_gl_sframe(sampled_frame->value, sampled_gl_sframe));
+
+  tc_groupby_aggregator_destroy(gb_manual);
+  tc_flex_list_destroy(column_list);
+  tc_sframe_destroy(sf);
+  tc_sframe_destroy(sampled_frame);
 }
 
-BOOST_AUTO_TEST_SUITE_END()
+
+BOOST_AUTO_TEST_CASE(test_sframe_groupby_random_sframe) {
+
+  tc_error* error = NULL;
+  std::string all_types = "RZSVLD"; 
+  // R: real, Z: integer, S: string, V: vector, L: list, D: double
+  tc_groupby_aggregator* gb1 = new_tc_groupby_aggregator();
+  tc_flex_list *column_list1 = new_tc_flex_list();
+  
+  TS_ASSERT(error == NULL);
+  size_t n_rows1 = 10000;
+  size_t n_columns1 = 100;
+  std::string column_types1 = "R";
+  for (int index = 1; index < n_columns1; index++) {
+    column_types1 += all_types[rand() % 6];
+    srand(time(NULL));
+  }
+  size_t seed = rand();
+  turi::gl_sframe sf_gl1 = _generate_random_sframe(n_rows1, column_types1, seed,
+    false, 0.0); 
+  tc_sframe *sf1 = new_tc_sframe(sf_gl1);
+  turi::gl_sarray column;
+  std::string column_name, zeroth_column, last_column;
+  zeroth_column = sf_gl1.column_name(0);
+  last_column = sf_gl1.column_name(n_columns1-1);
+  
+  TS_ASSERT(sf1->value.num_columns() == sf_gl1.num_columns());
+  TS_ASSERT(sf1->value.column_names() == sf_gl1.column_names());
+  TS_ASSERT(sf1->value.column_types() == sf_gl1.column_types());
+  TS_ASSERT(check_equality_gl_sframe(sf1->value, sf_gl1));
+
+  // C interface
+  tc_groupby_aggregator_add_sum(gb1, "a_sum", zeroth_column.c_str(), &error);
+  TS_ASSERT(error == NULL);
+  tc_groupby_aggregator_add_count(gb1, "a_count", &error);
+  TS_ASSERT(error == NULL);
+
+  tc_flexible_type* last_ft = tc_ft_create_from_cstring(last_column.c_str(), 
+    &error);
+  TS_ASSERT(error == NULL);
+  tc_flex_list_add_element(column_list1, last_ft, &error);
+  TS_ASSERT(error == NULL);
+
+  tc_sframe* sampled_frame1 = tc_sframe_group_by(sf1,column_list1,gb1,&error);
+  TS_ASSERT(error == NULL);
+
+  // C++ interface
+  std::vector<std::string> group_keys;
+  group_keys.push_back(last_column);
+  std::map<std::string, turi::aggregate::groupby_descriptor_type> operators;
+  operators.insert({"a_sum", turi::aggregate::SUM(zeroth_column)});
+  operators.insert({"a_count", turi::aggregate::COUNT()});
```

Comment:
This looks great!  If we can fill out something simple for each of the operators, that would be great.  

To make that easier, I'm fine with just copy-pasting this code and replacing the SUM and COUNT with each of the other operators.  

Thanks!